MoMA specific changes

Hwesta opened the pull request:
Changes from the MoMA private repo, squashed into a handful of commits. This is likely to be broken into several PRs as each feature gets merged. 

Changes:
- Add DRMC README
- copyToNetX client script, for after uploading to AtoM/DRMC
- New config page for NetX related config
- Normalization job displays red if any normalization fails
- Overload the accession number as the TMS object ID and use it as the AtoM/DRMC slug
- Add LDAP support

TODO:
- [x] Check if FPR changes are already in QA - yes, commit dropped
- [x] Test DRMC upload & copy to NetX script
- [ ] Make displaying the NetX config conditional (like DSpace transfer type) - not needed with #517 ?
- [x] Allow passing in a DRMC slug properly, instead of overloading the accession number field - uses new access system identifier field
- [ ] Allow the accession # field to have a custom label - not needed with access system ID?
- [x] Strip private information from LDAP implementation and merge to qa
- [x] Make DIP upload to AtoM/Binder/NetX more generic - new chain for upload to DRMC that includes NetX
- [x] Convert SQL migration to Django migration (from #463)

mcantelon committed:
Added config notes in README.DRMC.

mcantelon committed:
Client scripts: add copyToNetX

commit c8a68d93ce9145733267f7e3ef0aaa0764c2e2b5
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Tue Jan 20 13:05:54 2015 -0800

    NetX now appends to metadata file. refs #7797

commit 0c68ef56988c5db141d9a2cd0d59bfb550069146
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Jan 7 16:56:06 2015 -0800

    Added new way to specify component ID. refs #7772

    Added support for specifying TMS component ID, passed to NetX via
    a CSV file, using a JSON file in a transfer's metadata directory.

commit 7f71b3de239e22a10088220c3e2e03277a36ae9f
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Fri Aug 15 17:06:22 2014 -0700

    Minor change to NetX CSV render.

commit 65c7c6b74fd94dadc5b34d6d0a3115a0491fa828
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Thu Aug 14 14:30:16 2014 -0700

    Minor changes to NetX CSV render.

commit 8bee589b1bbb9078ebe4c4c55634dcc359fac8b0
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Tue Aug 12 13:42:30 2014 -0700

    Adjustments to NetX copy CSV output.

commit 71574a14d18c77ceba85c7ac16586ecba9246186
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Jul 9 12:48:14 2014 -0700

    MoMA-requested change to Netx copy.

commit 8085d9002059aecbf7234bf55fc1dafbc71ddbe5
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Thu Jun 12 19:11:08 2014 -0700

    Component ID no longer required for NetX copy.

commit 7109a24e79a19cb0490bae8a38f64b05de164c93
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Mon May 26 15:36:45 2014 -0700

    Changed how NetX copy works based on feedback.

commit d85acccf4dbb9702dc52a7ad0e6ca31f2ce9edd0
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Mon May 26 01:11:13 2014 -0700

    Added component ID to NetX metadata CSV.

commit d032c268229f9af9e1604bd73dc0e046aca6fe97
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Thu May 22 22:34:33 2014 -0700

    Added object ID to NetX metadata CSV.

commit 7976e05dde34ce5b687769191b3872289ea23435
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Thu May 22 19:17:52 2014 -0700

    Roughed in sidecar metadata writing.

commit e1427293c532f1dc96b719521d8f4351c300cafd
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Thu May 22 17:32:43 2014 -0700

    Changed copy to NetX so only objects get copied.

commit eb5e4b6a5242a023820a332bdd718d51a8d004bb
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Sun May 18 18:42:49 2014 -0700

    Added NetX copy script.

    Added a script to copy DIP files to the NetX directory.

mcantelon committed:
SQL: DRMC changes

commit 7407ab442b02fc108a946400f217d1c296e28c9e
Author: Misty De Meo <mistydemeo@gmail.com>
Date:   Fri Jun 20 11:58:20 2014 -0700

    Normalize: show all failures in ingest tab

    Fixes #6845.

    Conflicts:

        src/MCPServer/share/mysql_dev1.sql

commit db69dd326fdeeb647c67004a0cd3307d38cf4ccd
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Thu May 22 16:01:49 2014 -0700

    Add optional copy to NetX to workflow.

commit b216d2f06d2f3ff0b586c06c31d2f219b060e318
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Mon Mar 24 16:55:44 2014 -0700

    Committed missing file.
commit b60284a4c71b9425a4ac9faeef2c7d2dca8c56a5
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 19 11:21:09 2014 -0700

    Added DB tweak.

mcantelon committed:
Dashboard: Add NetX config

commit 7109a24e79a19cb0490bae8a38f64b05de164c93
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Mon May 26 15:36:45 2014 -0700

    Changed how NetX copy works based on feedback.

commit e6580d2be95b83514bd6664e8e091e4970c409b4
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Mon May 12 17:51:04 2014 -0700

    Added NetX admin page.

mcantelon committed:
Dashboard: Use accession ID for TMS ID

commit 25fd4e8d39039262f79eab0ebd01b44560dc5341
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Tue Dec 9 18:15:11 2014 -0800

    Fixed accession ID lookup failure. #refs 7660

    Accession ID lookup, used to pass ID in JSON feed, failing
    intermittantly due to transfer UUID not always being set in
    file records.

commit 053cbc91ad6da2acb722a0daf74e5f07fe5e6e17
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 12 13:53:26 2014 -0700

    Fixed issue with ingest status.

commit 3dc6c80b7d041c6523a7c27d5cd7c42ed5ee9157
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Tue Mar 4 23:45:48 2014 -0800

    Embed accession ID in status data (for DRMC customization).

commit 2d7efa7fcdcad0f5a337a5e41b884c6079decf87
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 19 23:46:30 2014 -0700

    Fixed issues with DRMC DIP target prefix selection.

commit 5bdee041818446d05832299619c26a30bb9f1ce4
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 19 16:22:41 2014 -0700

    Added DIP target prefix selection.

commit 4ef403c45e760882dc582dbb602190e69f3837e6
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 19 11:31:25 2014 -0700

    Another text tweak.

commit d2c534098b7a5e43a53f3bc7425494d2677dac20
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 19 11:25:10 2014 -0700

    Minor text tweak.

commit 4b5e2d3d6def3d247a975fc519cac893a5793aff
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Tue Mar 4 16:48:48 2014 -0800

    Changed label.

mcantelon committed:
JS: automatically fill in AtoM upload form

commit 2d7efa7fcdcad0f5a337a5e41b884c6079decf87
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 19 23:46:30 2014 -0700

    Fixed issues with DRMC DIP target prefix selection.

commit 5bdee041818446d05832299619c26a30bb9f1ce4
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 19 16:22:41 2014 -0700

    Added DIP target prefix selection.

commit 884b8083524f96a15783d8cbd141c14e4567f4ef
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 12 14:00:46 2014 -0700

    Added use of access # for AtoM DIP slug.

commit 7c18cf8b2fbf2b08dcfba4a18346d68b629cef2f
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Wed Mar 5 00:38:38 2014 -0800

    Only ask for DIP upload slug if no accession number.

mistydemeo commented on src/MCPClient/lib/clientScripts/copyToNetX.py:
```diff
@@ -0,0 +1,170 @@
+import shutil
+import sys
+
+sys.path.append("/usr/lib/archivematica/archivematicaCommon")
```
Can probably remove this now?

mistydemeo commented:
The contents of `src/MCPServer/share/mysql_fpr_dev1.sql` made it into Archivematica in another commit; you can omit those.

mistydemeo commented on src/dashboard/src/templates/transfer/grid.html:
```diff
@@ -191,7 +191,7 @@
           <div style="float:left;">
             <input type='text' id='transfer-accession-number'></input>
-            <div class="help-block">Accession no.</div>
+            <div class="help-block">TMS Object ID</div>
```
We probably don't actually want this change, right?

Hwesta replied:
Correct - this branch is a candidate to be the new MoMA deployed branch, and also a discussion place for how these things can be merged (or not) into qa. Accession number vs object id as slug for DIP upload is one of the things we'd have to work out.

jhsimpson replied:
In this branch, we do need this change, so it can be tested.  We talked about refactoring the way the feature related to this line is implemented, before merging (see the 'Allow passing in a DRMC slug properly, instead of overloading the accession numebr field' checkbox in the list on the PR description

Hwesta commented:
> The contents of src/MCPServer/share/mysql_fpr_dev1.sql made it into Archivematica in another commit; you can omit those.

Good to know, I'll remove them.

mistydemeo commented on src/dashboard/src/templates/ingest/grid.html:
```diff
@@ -234,13 +234,16 @@
     </div>
     <div class="modal-body">
-      <p>Upload the generated DIP to AtoM using the permalink of the target description.</p>
+      <p>Upload the generated DIP to DRMC using a TMS object ID or a technology record slug.</p>
```
Ditto, we don't actually want to grab this commit, right?

jhsimpson replied:
For this dev branch, we do want this commit, but again we need some additional dev work to refactor this for a more generic approach.  Something like providing a way in the admin page to set this message, or a config file, perhaps.

mistydemeo commented on src/dashboard/src/templates/administration/sidebar.html:
```diff
@@ -36,6 +37,8 @@
 
     <li class="{% active request url_atk_dips %}"><a href="{{ url_atk_dips }}">Archivists Toolkit DIP upload</a></li>
 
+    <li class="{% active request url_netx %}"><a href="{{ url_netx }}">NetX</a></li>
```
Maybe hide this by default, the way you can hide the AtoM DIP upload link?

Hwesta replied:
Likely to happen - see TODO item 2. :)

Hwesta committed:
Migration: DRMC changes

Hwesta committed:
copyToNetX: Update to use ORM

Hwesta committed:
Admin: Fix NetX config styling

Hwesta committed:
Migration: Separate Upload to Binder & Upload to AtoM

Create different chains for uploading to Binder (including Upload to NetX)
and uploading to AtoM.

Hwesta committed:
WIP Processing Config: Add upload & store DIP config options

This doesn't always work because of issues around SIP vs DIP directory.

Hwesta committed:
Update transfer browser webpack

Hwesta committed:
Models: Remove unused Transfer fields

Hwesta committed:
Transfer: Store access system ID

Allow an access system ID to be submitted when a transfer is created. This
can be used later to help automate submitting a DIP to an access system.

Hwesta committed:
Cleanup

Hwesta committed:
Upload to AtoM: Use access_system_id instead of accession no

When uploading a DIP to AtoM, use the new access_system_id instead of the
overloaded accession no.

Hwesta committed:
Add LDAP support

Squashed commit of the following:

commit 727365896fda018061d721bc50d82331113c7d07
Author: Holly Becker <hbecker@artefactual.com>
Date:   Thu Nov 17 12:12:08 2016 -0800

    Dashboard: Remove manual activation of LDAP accounts

commit 209cb58eb22cf680c4beded3e392a56a9c538083
Author: Holly Becker <hbecker@artefactual.com>
Date:   Thu Nov 17 11:57:57 2016 -0800

    LDAP Settings: Remove MoMA-specific comments

commit becd2ee0cd6639b5b8ca5fc09eef9df14c200428
Author: Holly Becker <hbecker@artefactual.com>
Date:   Thu Oct 27 13:43:27 2016 -0700

    Settings: Add LDAP config

commit a9e9743ab94c5d03f7837b6419c8e0b2990efc60
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Sat Dec 6 15:25:10 2014 -0800

    Fix for list LDAP users with missing attributes.

    If a user is missing an attribute, "mail" for example, in the LDAP
    directory a KeyError was causing the user admin web page to error
    before returning.

commit b316e55fdd598a2c9d56a17a3fa0398e542fac8b
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Thu Jun 5 23:48:11 2014 -0700

    Adding LDAP dashboard deps.

commit ae261ebec1cf533ae0c7997fc5ab95dad4e4c2bc
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Tue May 6 18:36:05 2014 -0700

    Added logic to keep out unapproved LDAP users.

commit ce36213f412ca08a808d266ce119a2819f0086e3
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Tue May 6 18:35:21 2014 -0700

    Added ability to administer users from LDAP.

commit 63e3d3b2896b41c94bc15eacd134e6e2ab9493ee
Author: Mike Cantelon <mcantelon@gmail.com>
Date:   Sun May 4 15:19:15 2014 -0700

    Roughed in LDAP authentication.

Hwesta commented:
This should be complete now, and should be broken up into 4 pull requests for ease of code review:
- NetX support - upload to & configure NetX, new upload to Binder DIP upload option.
- Access system ID - allow an access system slug to be specified on transfer creation
- LDAP - Add settings file for LDAP support
- Transfer cleanup - remove unused field in Transfer table

The README.DRMC can also be dropped, since the configuration settings can be stored elsewhere.

Hwesta committed:
copyToNetX: handle multiple Jobs when fetching DIP path

Since we're uploading to multiple systems, there are multiple Upload DIP
jobs, all with the same path. Take the first one instead of erroring.

Hwesta committed:
Migrations: Handle parallel migrations

Hwesta commented:
Rebased on top of qa/1.x, now includes #554 as well.

Hwesta commented:
This also needs the changes in https://github.com/artefactual-labs/transfer_browser/pull/13

sevein commented:
The PR https://github.com/artefactual-labs/ansible-archivematica-src/pull/29 in ansible-archivematica-src has been closed. The dependencies can be added to osdeps.

sromkey commented:
We're tracking the changes needed for MoMA in different methods now so I'm closing this PR :)
